### PR TITLE
Feature/stagemode stage refactoring 1878

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

File: src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import * as styled from './Stage.styled';
+
+export interface StagePropsInterface {
+  onRemoteUserClick?: (remoteUser: AgoraRemoteUserInterface, type: string) => void;
+}
+
+const Stage: React.FC<StagePropsInterface> = ({onRemoteUserClick}) => {
+  const [cols, setCols] = useState<string>('cols-1');
+
+  const {
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx:20]: inline pls

File: src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+              size="xl"
+              align="center"
+              transform="uppercase"
+              weight="bold"
+            />
+          </styled.MessageContainer>
+        )}
+
+        {stageModeStore.isOnStage && (
+          <styled.MediaPlayerContainer key="stageuser-local">
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx:60]: key?

File: src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+import {IconSvg, Text} from 'ui-kit';
+
+import * as styled from './Stage.styled';
+
+export interface StagePropsInterface {
+  onRemoteUserClick?: (remoteUser: AgoraRemoteUserInterface, type: string) => void;
+}
+
+const Stage: React.FC<StagePropsInterface> = ({onRemoteUserClick}) => {
+  const [cols, setCols] = useState<string>('cols-1');
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx:18]: useState after connection to store

File: src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+  const {
+    mainStore: {agoraStore},
+    sessionStore
+  } = useStore();
+  const {userDevicesStore, stageModeStore} = agoraStore;
+
+  const {t} = useTranslation();
+
+  const stageCount = useObserver(() =>
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx:28]: Looks like store function

File: src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+              currentUser={sessionStore.profile ?? undefined}
+              loadCurrentUserProfile={sessionStore.loadUserProfile}
+            />
+            <styled.RemoteUserActionsContainer>
+              <styled.RemoteUserActions>
+                <styled.RemoteUserActionButtons>
+                  {!user.isMuted && (
+                    <styled.RemoteUserActionButton
+                      onClick={() => {
+                        if (onRemoteUserClick) {
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx:92]: onRemoteUserClick?.(user, 'mute');

File: src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+                    >
+                      <styled.RemoteUserAction>
+                        <IconSvg name="microphoneOff" size="medium-large" />
+                        <Text text={t('actions.mute')} size="s" />
+                      </styled.RemoteUserAction>
+                    </styled.RemoteUserActionButton>
+                  )}
+                  <styled.RemoteUserActionButton
+                    onClick={() => {
+                      if (onRemoteUserClick) {
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx:105]: onRemoteUserClick?.(user, 'remove');

File: src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx [hunk 7]
```diff
@@ -0,0 +1,125 @@
+import React, {useEffect, useState} from 'react';
+import {observer, useObserver} from 'mobx-react-lite';
+import cn from 'classnames';
+import {useTranslation} from 'react-i18next';
+
+import {useStore} from 'shared/hooks';
+import {AgoraRemoteUserInterface} from 'stores/MainStore/models/AgoraStore/models';
+import {MediaPlayer} from 'scenes/collaboration/pages/StageModePage/components';
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx:8]: ui-kit component

File: src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+import {IconSvg, Text} from 'ui-kit';
+
+import * as styled from './Stage.styled';
+
+export interface StagePropsInterface {
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/organisms/Stage/Stage.tsx:13]: why export?

**Review by PAKovalev — CHANGES_REQUESTED** [review]

**Review by PAKovalev — APPROVED** [review]

**Review by aminbksi — APPROVED** [review]

**Pull request merged**